recipe/meta.yaml

```diff
@@ -26,9 +26,9 @@ build:
 requirements:
   build:
     - {{ compiler('c') }}
-    - yasm               # [win32]
+    - nasm
```

Comment:
Actually I think this is only a Windows requirement. In fact am now remembering this was [only needed/used on Windows 32-bit](  https://wiki.openssl.org/index.php/Compilation_and_Installation#W32_.2F_Windows_NT_-_Windows_9x ) as the selector notes.

There are some differences between support for 32-bit and 64-bit assembly on Windows. For example [there is no inline assembly support in VS for 64-bit]( https://docs.microsoft.com/en-us/cpp/build/intrinsics-and-inline-assembly?view=vs-2017 ). Not sure if this why [OpenSSL doesn't use assembly on Windows 64-bit]( https://wiki.openssl.org/index.php/Compilation_and_Installation#W64 ). Given this, maybe we should just drop the assembler requirement entirely. Thoughts?

Also sorry for the mix-up. 😞

Reply:
If we do decide to drop the assembler requirement, we can clean out some patches and other workarounds for YASM. ( https://github.com/conda-forge/openssl-feedstock/commit/cddd3f60d674258699a7853e99fb1a6d29cddef0 )

Reply:
@jakirkham, thanks for your comments! I have already removed YASM patches and workarounds, and following your suggestion I have entirely removed the assembler requirement.

I think that what we are saying is valid for 1.0.2p, but it seems from the official docs that things have changed in the newer versions. In fact, in `openssl`'s `master` branch, `nasm` is stated as a strict requirement.